dotCMS/src/integration-test/java/com/dotmarketing/factories/MultiTreeAPITest.java

```diff
@@ -0,0 +1,380 @@
+        tree = APILocator.getMultiTreeAPI().getMultiTree(PAGE, CONTAINER+ 0, CONTENTLET + 0, RELATION_TYPE+0);
+        assertTrue("multiTree reorders", tree.getTreeOrder()==4 );
+        APILocator.getMultiTreeAPI().saveMultiTree(tree.setTreeOrder(2));
+        List<MultiTree> list = APILocator.getMultiTreeAPI().getMultiTrees(PAGE, CONTAINER+0, RELATION_TYPE+0);
+        assertTrue("multiTree reorders", list.get(2).equals(tree));
+
+    }
+    
+    @Test
+    public  void findByChild() throws Exception {
+        deleteInitialData();
+        buildInitalData() ;
+        
+        List<MultiTree> list = APILocator.getMultiTreeAPI().getMultiTreesByChild(CONTENTLET + "0");
+        
+        assertTrue("getByChild returns all results", list.size() == runs );
+        
+        
+        
+    }
+    
+    
+    
+    @AfterClass
+    public static void deleteInitialData() throws Exception {
+
+        List<MultiTree> list = APILocator.getMultiTreeAPI().getMultiTrees(PAGE);
+
+        for(MultiTree tree : list) {
+            APILocator.getMultiTreeAPI().deleteMultiTree(tree);
+        }
+
+    }
+    
+    
+    
+    
+    
+    @Test
+    public  void testSaveMultiTree() throws Exception {
+        MultiTree mt = new MultiTree()
+                .setContainer(CONTAINER)
+                .setHtmlPage(PAGE)
+                .setContentlet("NEW_ONE")
+                .setTreeOrder(0)
+                .setInstanceId(RELATION_TYPE + 0);
+        
+        APILocator.getMultiTreeAPI().saveMultiTree(mt);
+        
+        MultiTree mt2 = APILocator.getMultiTreeAPI().getMultiTree(mt.getHtmlPage(), mt.getContainer(), mt.getContentlet(), mt.getRelationType());
+        assertTrue("multiTree save and get equals", mt.equals(mt2));
+    }
+    
+    
+    
+
+    
+    
+    
+    @Test
+    public void testLegacyMultiTreeSave() throws Exception {
+
+        
+        long time = System.currentTimeMillis();
+
+        
+        MultiTree multiTree = new MultiTree();
+        multiTree.setHtmlPage( PAGE+time);
+        multiTree.setContainer( CONTAINER +time);
+        multiTree.setContentlet( CONTENTLET +time);
+        multiTree.setTreeOrder( 1 );
+        APILocator.getMultiTreeAPI().saveMultiTree( multiTree );
+        
+        
+        MultiTree mt2 = APILocator.getMultiTreeAPI().getMultiTree(PAGE+time, CONTAINER +time, CONTENTLET +time, Container.LEGACY_RELATION_TYPE);
+        
+        assertTrue("multiTree save without relationtype and get equals", multiTree.equals(mt2));
+    }
+    
+    
+    @Test
+    public void testGetPageMultiTrees() throws Exception {
+        // GET ANY REAL PAGE (NO ES)
+        Map<String, Object> map = new DotConnect().setSQL("select contentlet_version_info.* from contentlet_version_info where working_inode in (select inode from contentlet where structure_inode in (select inode from structure where lower(velocity_var_name)='htmlpageasset'))").setMaxRows(1).loadObjectResults().get(0);
```

Comment:
`htmlpageasset` should be replaced by this constant https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPI.java#L71

Reply:
I used used datagens instead